numba/np/arraymath.py

```diff
@@ -656,6 +656,33 @@ def array_argmin_impl_generic(arry):
     return min_idx
 
 
+@register_jitable
+def array_nanargmin_impl_float(arry):
```

Comment:
The code here and in `array_nanargmax_impl_float` are practically the same. You can refactor it to something like:

```python
def build_array_nanarg_impl_float(name, operator):
    @register_jitable
    def array_nanargmax_impl_float(arry):
        if numpy_version < (1, 22) and arry.size == 0:
            if name == 'argmax':
                raise ValueError("attempt to get argmax of an empty sequence")
            else:
                raise ValueError("attempt to get argmin of an empty sequence")
        has_non_nan_index = False
        for v in arry.flat:
            if not np.isnan(v):
                max_value = v
                max_idx = 0
                has_non_nan_index = True
                break
        if not has_non_nan_index:
            raise ValueError("All-NaN slice encountered")

        idx = 0
        for v in arry.flat:
            if np.isnan(v):
                continue
            # if v > max_value:
            if operator(v, max_value):
                max_value = v
                max_idx = idx
            idx += 1
        return max_idx
    return array_nanargmax_impl_float
```

And inside the `@overload` function, do:

```python
    ...
    fn = build_array_nanarg_impl_float('argmax', operator.gt)  # or operator.lt for nanargmin
    if is_nonelike(axis):
        def array_argmax_impl(arr, axis=None):
            return fn(arr)
            # return array_nanargmax_impl_float(arr)
    else:
        array_argmax_impl = build_argmax_or_argmin_with_axis_impl(
            arr, axis, fn
        )
```